addons/l10n_it_edi/models/account_edi_format.py

```diff
@@ -100,36 +114,53 @@ def _l10n_it_edi_check_ordinary_invoice_configuration(self, invoice):
+            errors.append(_("Cannot apply Reverse Charge to a bill which contains both services and goods."))
+
         # <2.2.1>
         for invoice_line in invoice.invoice_line_ids:
             if not invoice_line.display_type and len(invoice_line.tax_ids) != 1:
-                raise UserError(_("You must select one and only one tax by line."))
+                errors.append(_("In line %s, you must select one and only one tax by line.", invoice_line.name))
 
         for tax_line in invoice.line_ids.filtered(lambda line: line.tax_line_id):
             if not tax_line.tax_line_id.l10n_it_kind_exoneration and tax_line.tax_line_id.amount == 0:
                 errors.append(_("%s has an amount of 0.0, you must indicate the kind of exoneration.", tax_line.name))
 
         return errors
 
+    def _l10n_it_edi_is_simplified(self, invoice):
+        buyer = invoice.commercial_partner_id
+        return all([
+            self.env.ref('l10n_it_edi.account_invoice_it_simplified_FatturaPA_export', raise_if_not_found=False),
+            not self._l10n_it_edi_is_self_invoice(invoice),
+            self._l10n_it_edi_check_buyer_invoice_configuration(invoice),
+            not buyer.country_id or buyer.country_id.code == 'IT',
+            buyer.l10n_it_codice_fiscale or (buyer.vat and (buyer.vat[:2].upper() == 'IT' or buyer.vat[:2].isdecimal())),
+            invoice.amount_total <= 400,
+        ])
+
     def _l10n_it_edi_check_simplified_invoice_configuration(self, invoice):
-        errors = self._l10n_it_edi_check_buyer_invoice_configuration(invoice)
-        template = self.env.ref('l10n_it_edi.account_invoice_it_simplified_FatturaPA_export', raise_if_not_found=False)
+        return [] if self._l10n_it_edi_is_simplified(invoice) else self._l10n_it_edi_check_buyer_invoice_configuration(invoice)
 
-        if errors and template:
-            buyer = invoice.commercial_partner_id
-            if ((not buyer.country_id or buyer.country_id.code == 'IT')
-                and (buyer.l10n_it_codice_fiscale or (buyer.vat and (buyer.vat[:2].upper() == 'IT' or buyer.vat[:2].isdecimal())))
-                and invoice.amount_total <= 400):
-                return []
+    def _l10n_it_edi_partner_in_eu(self, partner):
+        europe = self.env.ref('base.europe', raise_if_not_found=False)
+        country = partner.country_id
+        return not europe or not country or country in europe.country_ids
 
-        return errors
+    def _l10n_it_edi_services_or_goods(self, invoice):
+        lines = invoice.invoice_line_ids.filtered(lambda l: not l.display_type)
+        scopes = lines.mapped(lambda line: (line.tax_ids and set([x for x in line.tax_ids.mapped("tax_scope") if x])) or set([line.product_id and line.product_id.type]))
+        services_or_goods = set([y for x in scopes if x for y in x])
+        if services_or_goods == set(['consu', 'service']):
+            return "both"
+        return services_or_goods and services_or_goods.pop() or "both"
```

Comment:
The tax grids for reverse charge are separated between those only for goods or only for services.
If the invoice has both services and goods, or if the lines and the taxes don't specify them, then we have a problem.
Your code is clearer indeed.

Reply:
No, wait, it doesn't work as it is... `line` not defined.
I've put this. I'm now taking "consu" as a default for lines that I cannot determine, so it won't be `both` if not specified.
It's probably safer to assume that we're selling goods.
```py
    def _l10n_it_edi_services_or_goods(self, invoice):
        """ 
            Services and goods have different tax grids when VAT is Reverse Charged, and they can't
            be mixed in the same invoice, because the TipoDocumento depends on which which kind
            of product is bought and it's unambiguous.
        """
        lines = invoice.invoice_line_ids.filtered(lambda l: not l.display_type)
        scopes = lines.mapped(lambda line: line.tax_ids.filtered(lambda x: x.tax_scope).mapped('tax_scope') or line.product_id.type or 'consu')
        if set(scopes) == set(['consu', 'service']):
            return "both"
        return scopes and scopes.pop()
```